sycl/doc/extensions/experimental/sycl_ext_matrix/sycl_ext_intel_matrix.asciidoc

```diff
@@ -203,6 +203,147 @@ In the case of `ext_intel_packed` matrix memory layout, `row` and
 `col` represent the coordinates in the logical matrix before VNNI
 transformation.
 
+=== Load/Store/Fill With Out-of-Bounds Checks
+`joint_matrix_load`, `joint_matrix_store`, and `joint_matrix_fill`
+operations do not perform bounds checking. When the global matrix size is
+not multiple of the joint matrix size, this extension adds a new form
+of API to load, store, and fill joint matrix while checking the
+bounds. For load and fill, the out-of-bounds elements are set to
+0. For the store, they are dropped out.
+
+These functions are similar to the existing ones without bounds
+checking, namely `joint_matrix_fill`, `joint_matrix_load`, and
+`joint_matrix_store`. But they are different in three ways:
+
+* The pointer `base_src` or `base_dest` designates the base pointer of
+the original memory matrix excluding any offset calculations. The
+coordinates are given separately here.
+* The coordinates `coordX` and `coordY` to calculate the  pointer
+offset to load/store are given as separate arguments.
+* These variants take extra arguments to determine the global bounds
+`Height` and `Width` of the two-dimensional region to load to or store
+from.
+
+The two-dimensional memory region is interpreted as follows, where
+`Height` is number of rows in the memory matrix, `Width` is number of
+columns in the memory matrix, `stride` is number of rows that include
+the out of bounds data (depicted as x here).
+
+|<-----------`Width`-------->|
+
+|DDDDDDDDDDDDDDDDDxxx|
+
+|DDDDDDDDDDDDDDDDDxxx|
+
+|......................................................| `Height`
+
+|DDDDDDDDDDDDDDDDDxxx|
+
+|DDDDDDDDDDDDDDDDDxxx|
+
+|<------------`Stride`---------->|
+
+Note that joint matrix shape `Rows` and `Cols` represents a sub-block
+of the picture above for each sub-group execution. The out of bounds
+functions checks that the 2d block starting at the address to be
+accessed `[0, src]` and ending at `[Cols, src + Rows * stride]` does
+not exceed the bigger 2d block starting at `[0, src]` and ending at
+`[Width, src + Height * stride]`. For instance, load of `sub_b`
+defined as `joint_matrix<sub_group, bfloat16, use::b, 2, 3,
+layout::row_major> sub_b;` from `ptr + 9` checks that `ptr + 9` does
+not exceed `ptr + 10` where 10 is `Width` so it reads only 1 element
+for that iteration. The 2 columns left that exceed the total `10`
+columns are filled by zero.
+
+```c++
+namespace sycl::ext::intel::experimental::matrix {
+
+template <typename Group, typename T, size_t Rows, size_t Cols,
+          use Use, layout Layout, typename Tv>
+void joint_matrix_fill_checked(Group g, joint_matrix<Group, T, Use, Rows,
+          Cols, Layout> &m, Tv v, size_t Stride, size_t Height, size_t
+          Width, size_t coordX, size_t coordY);
```

Comment:
@MrSidims, 
In the load/store, the check does the following (let's just consider one horizontal dimension to simplify):
if (base_ptr + coordY + Cols > base_ptr+Width) --> out of bounds

Since we intend  to do the check just like load and store (just considering base_ptr=0), we need to add these arguments: Height, Width, Stride, CoordX, and CoordY. The check will do the following:
if (coordY + Cols > Width) --> out of bounds